### ISSUE TITLE
On timeout in client, the error code is operation_aborted instead of the more appropriate asio::error::timed_out

### ISSUE THREAD

**Issue by andreig1978** [eidheim/Simple-Web-Server#225]

**Comment by eidheim** [comment]
The `operation_aborted` error code is generated by the [`socket::cancel`](https://www.boost.org/doc/libs/1_67_0/doc/html/boost_asio/reference/basic_stream_socket/cancel/overload2.html) function.

**Comment by andreig1978** [comment]
Yes I know, but from the client's perspective it's a little confusing. I'll try to address it and submit a pull request.
